lib/stanford-mods/date_parsing.rb

```diff
@@ -8,10 +8,59 @@ module Mods
     #       or have an initialize method so we don't have to keep passing the date_str argument
     class DateParsing
 
-      # looks for dddd pattern in String and returns it if found
+      # get single facet value for date, generally an explicit year or "17th century" or "5 B.C."
+      # @param [String] date_str String containing a date (we hope)
+      # @return [String, nil] String facet value for year if we could parse one, nil otherwise
+      def self.facet_string_from_date_str(date_str)
+        # B.C. first in case there are 4 digits, e.g. 1600 B.C.
+        return facet_string_for_bc(date_str) if date_str.match(BC_REGEX)
+        # most date strings have a four digit year
+        result = sortable_year_for_yyyy(date_str)
```

Comment:
there's a comment at the top to pursue that when it is split off into a gem;  Jessie made the same suggestion.  I'm inclined to try to get this to the point of fixing Feigenbaum for now, rather than refactor it at this time.   I will expand on the comment a bit.